### DISCUSSION TITLE
Filter based on two fields

### DISCUSSION THREAD

**Discussion by Voldemorten** [directus/directus#20441] [Feature Requests]
### Summary

It would be awesome to have the ability to filter on two fields. Both in the app and in the api.

### Basic Example

We have a field called `buy_price` and a field called `sales_price`. To get all items with a `buy_price` greater than `sales_price`. We could use the filter `Buy Price is greater than Sales Price` or in json: 
```
filter: {
  "buy_price": {
    "_gt": "sales_price"
  }
}
```

### Motivation

It's a basic filtering addition, which gives so much more possibilities.

### Detailed Design

I don't know the best way to do it, but it could be something like:
App
- Normally we choose a field, and then an operator and then enter a value or `is empty` or something like that. Instead we could choose a field, then choose an operator but operator could be changed to `greater than or equal field` instead of `greater than or equal` and then the third option would be to choose a field. 

API:
- Solution could be something like testing on the operator, and if it's a certain "field" operator, then the value is interpreted as a field. For instance: 
```
filter: {
  "buy_price": {
    "_fgt": "sales_price"
  }
}
```
And choose a prefix for the operator that makes sense:`_gtf` (greater than field) 

### Must Have:
- Additional operators that searches for fields instead of values. So fields should be evaluated before they are passed to the query.
- Include these operators in the app, and figure out a nice way of displaying them.

### Drawbacks

It takes time to introduce new operators, and more time to code something efficiently that evaluates fields, and makes sure that they work correctly in the existing filtering. 

### Alternatives

Creating your own extension.

### Adoption Strategy

It is not a breaking change. We don't have to change anything, we just have to introduce something new.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
